File: app/src/main/java/au/id/blackwell/kurt/lantv/resolver/StaticMediaResolver.java
package au.id.blackwell.kurt.lantv.resolver;

import au.id.blackwell.kurt.lantv.MediaDetails;

/**
 * Immediately provides the media that it was constructed with.
 */
final class StaticMediaResolver implements MediaResolver {
    private MediaDetails mDetails;

    public StaticMediaResolver(MediaDetails details) {
        mDetails = details;
    }

    @Override
    public void resolve(Callback callback) {
        callback.onMediaResolved(mDetails);
    }

    @Override
    public void cancel(Callback callback) {
        // Nothing to cancel, this class is not asynchronous.
    }
}

File: app/src/main/java/au/id/blackwell/kurt/lantv/resolver/StaticMediaResolver.kt
package au.id.blackwell.kurt.lantv.resolver

import au.id.blackwell.kurt.lantv.player.MediaDetails

/**
 * Immediately provides the media that it was constructed with.
 */
internal class StaticMediaResolver(private val mDetails: MediaDetails) : MediaResolver {

    override fun resolve(callback: MediaResolver.Callback) {
        callback.onMediaResolved(mDetails)
    }

    override fun cancel(callback: MediaResolver.Callback) {
        // Nothing to cancel, this class is not asynchronous.
    }
}
